zebra-network/src/address_book.rs

```diff
@@ -597,6 +588,26 @@ impl AddressBook {
     }
 }
 
+impl PeerObserver for AddressBook {
+    fn recently_live_peers(&self, now: chrono::DateTime<Utc>) -> Vec<MetaAddr> {
+        let _guard = self.span.enter();
+
+        self.by_addr
+            .descending_values()
+            .filter(|peer| peer.was_recently_live(now))
+            .cloned()
+            .collect()
+    }
+}
+
+impl PeerObserver for Arc<Mutex<AddressBook>> {
+    fn recently_live_peers(&self, now: chrono::DateTime<Utc>) -> Vec<MetaAddr> {
+        self.lock()
+            .expect("mutex should be unpoisoned")
```

Comment:
Nitpick: This is a bit more meaningful for users, and developers who aren't familiar with the Mutex API.

```suggestion
            .expect("panic in a previous thread that was holding the mutex")
```